From fab3a057ccfa96db7832dd03ec227bbb751cd2e8 Mon Sep 17 00:00:00 2001
From: Appana Durga Kedareswara rao <appana.durga.rao@xilinx.com>
Date: Tue, 19 Mar 2019 14:28:26 +0530
Subject: [PATCH 33/40] net: can: xilinx_can: Fix incorrect assignment of
 bittiming_const for canfd

commit 0582f312842f6fc193e383865dbdbf6e3df963cb from
https://github.com/Xilinx/linux-xlnx.git

commit 9e5f1b2 ("can: xilinx_can: add support for Xilinx CAN FD core")
had wrong assignment of bittiming_const for axi canfd, It should point to
xcan_bittiming_const_canfd structure instead of xcan_bittiming_const_can.

This patch fixes this issue.

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.rao@xilinx.com>
Reviewed-by: Shubhrajyoti Datta <Shubhrajyoti.datta@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/can/xilinx_can.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/can/xilinx_can.c b/drivers/net/can/xilinx_can.c
index 1ca28ec3f844..6e6d133247ae 100644
--- a/drivers/net/can/xilinx_can.c
+++ b/drivers/net/can/xilinx_can.c
@@ -1640,7 +1640,7 @@ static const struct xcan_devtype_data xcan_canfd_data = {
 		 XCAN_FLAG_RXMNF |
 		 XCAN_FLAG_TX_MAILBOXES |
 		 XCAN_FLAG_RX_FIFO_MULTI,
-	.bittiming_const = &xcan_bittiming_const,
+	.bittiming_const = &xcan_bittiming_const_canfd,
 	.btr_ts2_shift = XCAN_BTR_TS2_SHIFT_CANFD,
 	.btr_sjw_shift = XCAN_BTR_SJW_SHIFT_CANFD,
 	.bus_clk_name = "s_axi_aclk",
-- 
2.17.1

